### PR TITLE
chore(v2): upgrade to Babel 7

### PR DESCRIPTION
**Pull request by fiennyangeln**
## Motivation

#984 
(Write your motivation here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

## Test Plan
1. Go to v2 folder
2. Run yarn && yarn test
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Related PRs
None
(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: v2/lib/webpack/base.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ module.exports = function createBaseConfig(props, isServer) {
       .loader('babel-loader')
       .options({
         babelrc: false,
-        presets: ['env', 'react'],
+        presets: ['@babel/env', '@babel/react'],
         plugins: [isServer ? 'dynamic-import-node' : 'syntax-dynamic-import'],
```
Comment (endiliey) [inline v2/lib/webpack/base.js]: Haven't really tried this out but i think the dynamic import node babel plugin is not compatible with babel 7.

See https://github.com/airbnb/babel-plugin-dynamic-import-node/pull/64

Bappening in CRA as well

https://github.com/facebook/create-react-app/pull/4984

File: v2/test/loadSetup.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import '@babel/polyfill';
```
Comment (endiliey) [inline v2/test/loadSetup.js]: Will it fails without this ? O.O

Comment (fiennyangeln) [reply v2/test/loadSetup.js]: yes!

File: v2/lib/webpack/base.js (hunk 1 above)
Comment (yangshun) [reply v2/lib/webpack/base.js]: @fiennyangeln Can you verify this? What do we do about this then?

**Comment by yangshun** [comment]
@fiennyangeln Let us know when this is good to review again.

File: v2/lib/webpack/base.js (hunk 1 above)
Comment (fiennyangeln) [reply v2/lib/webpack/base.js]: @yangshun yes, the PR for babel 7 upgrade for that plugin is still open. How about using https://www.npmjs.com/package/babel-plugin-transform-dynamic-import instead?

**Comment by fiennyangeln** [comment]
@yangshun please review 😄

**Comment by yangshun** [comment]
@fiennyangeln The tests are failing 😓

**Comment by fiennyangeln** [comment]
Ah sorry! I guess the changing of import order mess it all, reverted it back @yangshun

**Pull request merged**